Enable new perms syncer by default (#47783)
sourcegraph/sourcegraph: CHANGELOG.md

- Batch Changes: Log outputs from execution steps are now paginated in the web interface. [#46335](https://github.com/sourcegraph/sourcegraph/pull/46335)
- Monitoring: the searcher dashboard now contains more detailed request metrics as well as information on interactions with the local cache (via gitserver). [#47654](https://github.com/sourcegraph/sourcegraph/pull/47654)
- Renders GitHub pull request references in the commit list. [#47593](https://github.com/sourcegraph/sourcegraph/pull/47593)

> - Added a new background permissions syncer & scheduler which is backed by database, unlike the old one having an in-memory processing queue. The new system is enabled by default, but can be disabled and reverted to the old one using the feature flag `database-permission-sync-worker` set to `false`. 

Reviewer: ```suggestion
- Added a new background permissions syncer & scheduler which is backed by database, unlike the old one having an in-memory processing queue. The new system is enabled by default, but can be disabled and reverted to the old one using the feature flag `database-permission-sync-worker` set to `false`. [#47783](https://github.com/sourcegraph/sourcegraph/pull/47783)
```

> - Added a new background permissions syncer & scheduler which is backed by database, unlike the old one having an in-memory processing queue. The new system is enabled by default, but can be disabled and reverted to the old one using the feature flag `database-permission-sync-worker` set to `false`. [#47783](https://github.com/sourcegraph/sourcegraph/pull/47783)

Reviewer 2: ```suggestion
- Added a new background permissions syncer & scheduler which is backed by database, unlike the old one that was based on an in-memory processing queue. The new system is enabled by default, but can be disabled and reverted to the old one using the feature flag `database-permission-sync-worker` set to `false`. [#47783](https://github.com/sourcegraph/sourcegraph/pull/47783)
```

> - Added a new background permissions syncer & scheduler which is backed by database, unlike the old one having an in-memory processing queue. The new system is enabled by default, but can be disabled and reverted to the old one using the feature flag `database-permission-sync-worker` set to `false`. [#47783](https://github.com/sourcegraph/sourcegraph/pull/47783)

Reviewer 3: Improving on the suggestion from @mrnugget above: 
```suggestion
- Added a new background permissions syncer & scheduler which is backed by database, unlike the old one that was based on an in-memory processing queue. The new system is enabled by default, but can be disabled. Revert to the in-memory processing queue by setting the feature flag `database-permission-sync-worker` to `false`. [#47783](https://github.com/sourcegraph/sourcegraph/pull/47783)
```
